tests/fen_tests.rs

```diff
@@ -0,0 +1,350 @@
+    assert!(fen_util::valid_fen("7K/8/8/8/8/8/8/8 w - - 0 0"));
+    assert!(fen_util::valid_fen("7q/8/8/8/8/8/8/8 w - - 0 0"));
+    assert!(fen_util::valid_fen("7Q/8/8/8/8/8/8/8 w - - 0 0"));
+    assert!(fen_util::valid_fen("7n/8/8/8/8/8/8/8 w - - 0 0"));
+    assert!(fen_util::valid_fen("7N/8/8/8/8/8/8/8 w - - 0 0"));
+}
+
+#[test]
+fn test_invalid_fen() {
+    // Only positions
+    assert!(!fen_util::valid_fen(
+        "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR"
+    ));
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8"));
+    assert!(!fen_util::valid_fen("8/7k/8/8/8/8/8/8"));
+
+    // Missing parts
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 w - h8 0")); //move number
+    assert!(!fen_util::valid_fen("/8/8/8/8/8/8/8 w - h8 0 0")); // rank
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8 w - h8 0 0")); // rank
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 - h8 0 0")); // turn
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 w h8 0 0")); // castles
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 w - 0 0")); // en passant
+
+    // Invalid parts
+    assert!(!fen_util::valid_fen("9/8/8/8/8/8/8/8 w - a1 0 0")); // file out of range
+    assert!(!fen_util::valid_fen("0/8/8/8/8/8/8/8 w - a1 0 0")); // file out of range
+    assert!(!fen_util::valid_fen("7p1/8/8/8/8/8/8/8 w - a1 0 0")); // file out of range
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8/8 w - a1 0 0")); // extra rank
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 x - a1 0 0")); // invalid turn
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 wb - a1 0 0")); // two turns
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 w x a1 0 0")); // invalid castle
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 x kqx a1 0 0")); // invalid castles
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 w KQK a1 0 0")); // invalid castle
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 w - z1 0 0")); // invalid en passant
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 w - a0 0 0")); // invalid en passant
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 w - a9 0 0")); // invalid en passant
+    assert!(!fen_util::valid_fen("8/8/8/8/8/8/8/8 w - 9 0 0")); // invalid en passant
+    assert!(!fen_util::valid_fen("7x/8/8/8/8/8/8/8 w - a1 0 0")); // invalid piece
+}
+
+#[test]
+fn test_parse_en_passant() {
+    let mut cb = Chessboard::empty();
+
+    let passant = "-"; // empty
+    fen_util::parse_en_passant(&mut cb, passant);
+    assert!(cb.en_passant == 0);
+
+    let passant = "a1";
+    fen_util::parse_en_passant(&mut cb, passant);
+    assert!(cb.en_passant == 1);
+
+    let passant = "A1";
+    fen_util::parse_en_passant(&mut cb, passant);
+    assert!(cb.en_passant == 1);
+
+    let passant = "h8";
+    fen_util::parse_en_passant(&mut cb, passant);
+    assert!(cb.en_passant == 64);
+
+    let passant = "-"; // set it back to empty
+    fen_util::parse_en_passant(&mut cb, passant);
+    assert!(cb.en_passant == 0);
+
+    // INVALID
```

Comment:
What does this mean?

Reply:
An invalid en passant?

Reply:
In the context of `test_parse_en_passant` the invalid test means trying to have it parse something invalid. Ex A9 or H0